java/23ai-jdbc-vector-search/src/main/java/com/oracle/dev/jdbc/OracleAIVectorSearchWithJava.java

```diff
@@ -0,0 +1,196 @@
+        .prepareStatement(insertSql)) {
+      for (double[] vector : vectors) {
+        insertStatement.setInt(1, randomize());
+        insertStatement.setObject(2, vector, OracleType.VECTOR);
+        insertStatement.addBatch();
+      }
+      insertStatement.executeBatch();
+    }
+  }
+
+  private void retrieveVectorAsArray(Connection connection)
+      throws SQLException {
+    try (PreparedStatement queryStatement = connection
+        .prepareStatement(querySql);) {
+      System.out.println("SQL DML: " + querySql);
+      ResultSet resultSet = queryStatement.executeQuery();
+      double[] vector = null;
+      while (resultSet.next()) {
+        vector = resultSet.getObject(2, double[].class);
+      }
+      System.out.println("Retrieved VECTOR: " + Arrays.toString(vector));
+    }
+  }
+
+  private void retrieveVectorAsString(Connection connection)
+      throws SQLException {
+    try (PreparedStatement queryStatement = connection
+        .prepareStatement(querySql);) {
+      System.out.println("SQL DML: " + querySql);
+      ResultSet resultSet = queryStatement.executeQuery();
+      String vector = null;
+      while (resultSet.next()) {
+        vector = (String) resultSet.getObject(2);
+      }
+      System.out.println("Retrieved VECTOR: " + vector);
+    }
+  }
+
+  private void retrieveVectorWithBoundVector(Connection connection)
+      throws SQLException {
+    // Bind a Vector to a select
+    try (PreparedStatement queryStatement = connection
+        .prepareStatement(querySqlWithBind);) {
+      double[] inputVector = {1.0, 2.2, 3.3};
+      System.out.println("SQL DML: " + querySqlWithBind);
+      System.out.println("Bound VECTOR: " + Arrays.toString(inputVector));
+      queryStatement.setObject(1, inputVector, OracleType.VECTOR);
+      ResultSet resultSet = queryStatement.executeQuery();
+      resultSet.next();
+      double[] outputVector = resultSet.getObject(2, double[].class);
+      System.out.println("Retrieved VECTOR: " + Arrays.toString(outputVector));
+    }
+  }
+
+  private Connection getConnectionFromPooledDataSource() throws SQLException {
+    // Create pool-enabled data source instance
+    PoolDataSource pds = PoolDataSourceFactory.getPoolDataSource();
+    // set connection properties on the data source
+    pds.setConnectionFactoryClassName("oracle.jdbc.pool.OracleDataSource");
+    pds.setURL(URL);
+    pds.setUser(USERNAME);
+    pds.setPassword(PASSWORD);
+    // Configure pool properties with a Properties instance
+    Properties prop = new Properties();
+    prop.setProperty("oracle.jdbc.vectorDefaultGetObjectType", "String");
+    pds.setConnectionProperties(prop);
+    // Override any pool properties directly
+    pds.setInitialPoolSize(10);
+    // Get a database connection from the pool-enabled data source
+    Connection conn = pds.getConnection();
+    return conn;
+  }
+
+  private int randomize() {
+    return ThreadLocalRandom.current().nextInt();
```

Comment:
Please define ID as 
    NUMBER GENERATED ALWAYS as IDENTITY(START with 1 INCREMENT by 1)
and avoid using a random number for your primary key.

Reply:
Done as requested - added NUMBER GENERATED ALWAYS as IDENTITY(START with 1 INCREMENT by 1), 
modified the Java JDBC code, and removed the randomize() method. @Kuassim, please review; thanks.